### DISCUSSION TITLE
Is an API for the UPPAAL GUI provided?

### DISCUSSION THREAD

**Discussion by m5261136** [UPPAALModelChecker/UPPAAL-Meta#155] [Q&A] [answered]
The application I am building is going to handle time automata, can I use the UPPAAL GUI?

**Comment by mikucionisaau** [comment, marked as answer]
There are several APIs:
1) [UTAP](https://github.com/UPPAALModelChecker/utap/) library can be used to parse UPPAAL documents into C++ code.
2) `model.jar` (shipped with every UPPAAL distribution) can be used to read and create UPPAAL documents in Java code, see `demo/ModelDemo.java` example.
3) UPPAAL GUI can load a custom made plugin as a separate tab and then access document in the editor, call engine etc, see [uppaal-plugin-demo](https://github.com/mikucionisaau/uppaal-plugin-demo).
